learn.md

```diff
@@ -1,50 +1,262 @@
+
+The rest are generally not relevent to solving web challenges.
+
+### [Burp Suite](https://portswigger.net/burp/communitydownload)
+Burp allows you to intercept and edit requests coming from and going to your browser. 
+
+Burp has a rich suite of tools and extentions, but the tool with the most immediate relevency is the **Proxy** tab. Using Burp's internal browser (or your browser, although this requires  a bit of setup)
+
+When you open the browswer with the `Intercept is on` button enabled, you'll notice that your browswer pauses while loading and an HTTP request shows up in the Burp window:
+
+```http
+GET / HTTP/1.1
+Host: example.com
+Upgrade-Insecure-Requests: 1
+User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.125 Safari/537.36
+Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9
+Accept-Encoding: gzip, deflate
+Accept-Language: en-US,en;q=0.9
+Connection: close
+```
+
+This is the request the browser is trying to make. Hit the `Forward` button or turn the Intercept off and the browser will continue as normal.
+
+This functionality allows you to bypass interacting with the UI of a website and send data directly to the server. For example, if you were testing a login page that doesn't allow certain characters. Using Burp you could make a dummy request, intercept it, and add the illegal characters to the request
+
+```http
+POST /login HTTP/1.1
+Host: www.example.com
+Connection: close
+Content-Length: 73
+Accept: application/json, text/javascript, */*; q=0.01
+X-Requested-With: XMLHttpRequest
+User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.125 Safari/537.36
+Content-Type: application/x-www-form-urlencoded; charset=UTF-8
+Origin: www.example.com
+Sec-Fetch-Site: same-origin
+Sec-Fetch-Mode: cors
+Sec-Fetch-Dest: empty
+Accept-Encoding: gzip, deflate
+Accept-Language: en-US,en;q=0.9
+
+email=admin@example.com &password=%27AND 1=1;-- SQL injection, anyone?
+```
+
+For more information about using Burp, [check out PortSwigger's tutorial](https://portswigger.net/burp/documentation/desktop/getting-started) 
+
+# Reverse Engineering
+
+Reverse Engineering (Rev) challenges are generally logic puzzles (reversing crypto alto’s without the math), game hacking, crack me programs, flag checkers, homemade languages, and reversing obscure architectures. 
+
+Reverse engineering is generally the category with the least amount of people participating in it, this is partly due to difficulty and partly due to most people who are good at reverse engineering going to pwn. 
+
+## Tools:
+
+### Required Tools to start for free today
+
+- [Ghidra](https://ghidra-sre.org/)
+  - Use this to decompile binaries and executables
+- [GDB](gnu.org/software/gdb/)
+  - Useful for dynamic analysis
+- [Hex Editor](https://hexed.it/)
+  - Useful for editing binaries
```

Comment:
Can you add examples of basic use cases for each of these tools? See the Burp section for kind of what I'm looking for. If I installed Ghirda, what's one thing I can use immediately for pretty much anything? (previewing the disassembled C code comes to mind). 

Obviously you don't (and shouldn't) need to cover every scenario. If the tool has a lot of varied use, link to external tutorials or guides. 

Screenshots and code blocks are encouraged